doc/user_guide/io.rst

```diff
@@ -1027,11 +1033,279 @@ HDF5 file. All other keyword arguments will be passed to
+and stores floats,ints and strings as attributes.
+
+Nexus requires datasets have attributes so the structure
+of the metadata needs to be able to indicate the values and attributes of a 
+dataset. For example to store an indication of the units for an axis.
+
+Datasets and their attributes are defined through use of a   
+a defined metadata structure.
+The value of a dataset is set using a "value" keyword. 
+The attributes of a dataset are defined using a "attrs" dictionary. The 
+attributes of a group are defined using an "attrs" dictionary.
+The attributes can be set to define quantities such as units or 
+to implement Nexus attributes to conform to a Nexus format.
+
+Example of a Nexus monochromator format defined in metadata
+::
+    
+    ├── monochromator
+    │   ├── energy    
+    │   │   ├── value : 12.0
+    │   │   ├── attrs
+    │   │   │   ├── units : keV
+    │   │   │   ├── NXclass : NXmonochromator
+    
+Example of axis data stored with units
+::
+    
+    ├── axis_x
+    │   ├── value : array([1.0,2.0,3.0,4.0,5.0])
+    │   ├── attrs
+    │   │   ├── units : mm
+
+Examples of Group structure
+::
+
+    ├── instrument
+    │   ├── attrs    
+    │   │   ├── model : "JEOL"
+    │   ├── sample
+    │   │   ├── attrs
+    │   │   │   ├── name : "Standard"
+        
+
+When read in it is then consistent with the stored hdf format.
+
+.. code-block:: python
+
+    >>> orginal_metadata.monochromator.energy.value = 12.5
+    >>> orginal_metadata.monochromator.energy.attrs.units = "keV"
+
+so "value" and "attrs" keywords are requirements to access information but
+also to add additional structure to the metadata such as datasets.
+If a hyperspy signal is stored to nexus format and then loaded back
+the hyperspy metadata would need to be updated to reflect this structure.
+``metadata.Acquisition_instrument.SEM.beam_current`` would need to change to 
+``metadata.Acquisition_instrument.SEM.beam_current.value`` for example. 
+As this will likely break some specific hyperspy functions
+It's recommended that a "mapping" dictionary is passed to the load or
+file_reader to translate stored original_metadata to 
+hyperspy signal metadata.
+For example to map a hdf5 entry to hyperspy metadata.
+
+.. code-block:: python
+
+   >>> nexus_mapping = {"entry.instrument.beamline.DCM.dcm_energy":("Acquisition_instrument.TEM.beam_energy",None)}
+   >>> a = hs.load("test.nxs",mapping=nexus_mapping)
+      
+Hyperspy metadata is stored with "original_metadata" in the file
+Hyperspy metadata is not automatically restored when a signal is loaded
+from a Nexus file. A mapping dictionary is the preferred method to map
+file metadata to hyperspy metdata.   
+
+Inspecting
+^^^^^^^^^^
+Looking in a Nexus file for specific metadatda is often useful - .e.g to find
+what position a specific stage was at: 
+
+    >>> from hyperspy.io_plugins.nexus import get_metadata_in_file
+    >>> results=get_metadata_in_file("sample.hdf5",search_keys=["stage1_z"])
```

Comment:
To be more precise: ``read_metadata_from_file`` instead of ``get_metadata_in_file``?
Typo: remove `results=` to get the output displayed?

Reply:
Agreed and changed.